internal/connector/test/integration/features/connector-agent-api.feature

```diff
@@ -1160,9 +1160,72 @@ Feature: connector agent API
+    #--------------------------------------------------------------
+    # In this part of the Scenario we test admin connector deletion
+    #--------------------------------------------------------------
+    # create connector to test force delete
+    Given I am logged in as "Jimmy"
+    When I POST path "/v1/kafka_connectors?async=true" with json body:
+      """
+      {
+        "kind": "Connector",
+        "name": "example 2",
+        "namespace_id": "${connector_namespace_id}",
+        "channel":"stable",
+        "connector_type_id": "aws-sqs-source-v1alpha1",
+        "kafka": {
+          "id": "mykafka",
+          "url": "kafka.hostname"
+        },
+        "service_account": {
+          "client_id": "myclient",
+          "client_secret": "test"
+        },
+        "schema_registry": {
+          "id": "myregistry",
+          "url": "registry.hostname"
+        },
+        "connector": {
+            "aws_queue_name_or_arn": "test",
+            "aws_secret_key": "test",
+            "aws_access_key": "test",
+            "aws_region": "east",
+            "kafka_topic": "test"
+        }
+      }
+      """
+    Then the response code should be 202
+    And the ".status.state" selection from the response should match "assigning"
+    And I store the ".id" selection from the response as ${forced_connector_id}
+
+    # api admin should be able to see new connector deployment
+    Given I am logged in as "Ricky Bobby"
+    When I wait up to "10" seconds for a GET on path "/v1/admin/kafka_connector_clusters/${connector_cluster_id}/deployments" response ".total" selection to match "2"
+    Then I GET path "/v1/admin/kafka_connector_clusters/${connector_cluster_id}/deployments"
+    And the ".total" selection from the response should match "2"
+
+    # soft delete connector using admin API
+    When I DELETE path "/v1/admin/kafka_connectors/${forced_connector_id}"
+    Then the response code should be 204
+
+    # assigned connector and deployment should get marked for deletion
+    When I GET path "/v1/admin/kafka_connectors/${forced_connector_id}"
+    Then the ".desired_state" selection from the response should match "deleted"
+    When I GET path "/v1/admin/kafka_connector_clusters/${connector_cluster_id}/deployments"
+    Then the response code should be 200
+    And the ".total" selection from the response should match "2"
+
+    # force delete connector using admin API
+    When I DELETE path "/v1/admin/kafka_connectors/${forced_connector_id}?force=true"
+    Then the response code should be 204
+
+    # assigned connector and deployment should get deleted immediately
+    When I GET path "/v1/admin/kafka_connectors/${forced_connector_id}"
+    Then the response code should be 410
+    When I GET path "/v1/admin/kafka_connector_clusters/${connector_cluster_id}/deployments"
+    Then the response code should be 200
+    And the ".total" selection from the response should match "1"
```

Comment:
We should also check that the deployment related to the `forced_connector_id` is not more present in the response

Reply:
Do we also need a test to verify the secrets are wiped ?

Reply:
yeah, that's a good additional check.

Reply:
Secrets deletion verification is done in another connector delete test already, and force delete now reuses the existing Delete() func. Do we really need to verify that again?

Reply:
> Secrets deletion verification is done in another connector delete test already, and force delete now reuses the existing Delete() func. Do we really need to verify that again?

if it is done in another test, then it is fine.
Was unsure about the scope of this fix

Reply:
The admin soft delete and force delete implementation now reuses existing delete functions from the public API, just without the authz checks for connector ownership. The previous implementation duplicated _some_ of that code, hence the issue.